### ISSUE TITLE
malformed HTML causes a regex backtracking lockup

### ISSUE THREAD

**Issue by pravdomil** [probablyup/markdown-to-jsx#160]
Try execute following code in node.js v8.7.0 repl:

```js
const a =`<g>
<g>
<path fill="#ffffff"/>
</g>
<path fill="#ffffff"/>`

var HTML_BLOCK_ELEMENT_R = /^ *<([^ >/]+) ?([^>]*)\/{0}>(?=[\s\S]*<\/\1>)((?:[\s\S]*?(?:<\1[^>]*>[\s\S]*?<\/\1>)*[\s\S]*?)*?)<\/\1>\n*/

HTML_BLOCK_ELEMENT_R.exec(a)
```
node.js will hang at last command

**Comment by pravdomil** [comment]
inspiration

https://github.com/markdown-it/markdown-it/blob/c57f593b2395df0e12c1d7fb8c9cebb3f3920ff3/lib/common/html_re.js

https://github.com/commonmark/commonmark.js/blob/76de355590604d0375305c4b9ac790f169f98691/lib/common.js

**Comment by probablyup** [comment]
That isn’t valid HTML. There’s no second closing g block.

**Comment by pravdomil** [comment]
I have a valid svg on the begining but that's a section of it that is passed into `const parsed = rule.parse(capture, nestedParse, state);` during html blocks parsing.

https://github.com/probablyup/markdown-to-jsx/blob/1e3b1d5275a936ca219be8fa5ea6354d6984eda5/index.js#L402

**Comment by probablyup** [comment]
This should be handled by https://github.com/probablyup/markdown-to-jsx/pull/156/commits/14c26e2133211db10a7fae03ef3c389854d9038f

If you want to check out the deploy preview: https://deploy-preview-156--unruffled-neumann-76d9b9.netlify.com/

**Comment by pravdomil** [comment]
@probablyup great!

In meanwhile I created same parser as yours by using `markdown-it` and `htmlparser2`. I will publish it to GitHub soon.

**Comment by probablyup** [comment]
Fixed in master, will do a minor release soon.

**Comment by probablyup** [comment]
Released as 6.5.0
